lib/identity_cache/cached/association.rb

```diff
@@ -39,6 +39,42 @@ def inverse_name
         end
       end
 
+      def fetch(records)
```

Comment:
`fetch` now runs before every fetch type whereas before it only ran for `has_many`s and recursive `has_one`s. I don't see a problem with this.